### PR TITLE
SOF-132-Node-Ethernet-Class

### PR DESCRIPTION
**Pull request by ShaneTWilliams**
The purpose of this class is to encapsulate all basic UDP functionality. This class will eventually be used with a telemetry class more specific to our application that will construct and parse UDP packets, then send them using this class.

`main.cpp` provides an example for using this class - after instantiating and initializing a `UDPClass` object with `.init()`, you can receive data by calling `UDPClass::receivePacket()` and accessing the data at `UDPClass::iPacketRecvBuffer`. You can also send data by setting `UDPClass::iPacketSendBuffer` and calling `UDPClass::sendPacket()`.

### CODE REVIEW

File: OnPod/Node/Examples/Ethernet/ethernet.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+UDPClass::UDPClass(uint8_t ETHERNET_SS_PIN, IPAddress NODE_IP,
+                    uint16_t NODE_PORT, uint8_t NODE_TYPE):
+    ETHERNET_SS_PIN(ETHERNET_SS_PIN),
+    NODE_IP(NODE_IP),
+    NODE_PORT(NODE_PORT),
+    NODE_TYPE(NODE_TYPE)
+    {}
+
+void UDPClass::init() {
```
Comment (andrewnash) [inline OnPod/Node/Examples/Ethernet/ethernet.cpp:14]: Is there a reason to separate `init()` and the constructor? Maybe it's just a style thing I'm not aware of, either way, it's not a huge deal

**Review by andrewnash — APPROVED** [review]
Just 1 minor comment, besides that, I have no issues with this

File: OnPod/Node/Examples/Ethernet/ethernet.cpp (hunk 1 above)
Comment (ShaneTWilliams) [reply OnPod/Node/Examples/Ethernet/ethernet.cpp:14]: The goal of this was to quickly instantiate the global objects without any possibility of error.

Then, later in the code, you can call the `UDPClass::init()` function to take care of the more time-consuming calls. This gives you more control over the Node’s boot timing.

I think that makes sense? @stuckatmarine

**Comment by lwaghorn** [comment]
Good comments in the code. Looks good to me

File: OnPod/Node/Examples/Ethernet/ethernet.cpp (hunk 1 above)
Comment (andrewnash) [reply OnPod/Node/Examples/Ethernet/ethernet.cpp:14]: @stuckatmarine says I'm stupid and this is fine, don't mind me

File: OnPod/Node/Examples/Ethernet/main.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#include <EthernetUdp.h>
+#include "ethernet.h"
+
+// here as an example - this will be moved to a node struct header file
+enum NodeType {
+	REAR = 0,
+	BRAKE = 1,
+	LVDC = 2
+};
+const NodeType NODE_TYPE = REAR;
```
Comment (stuckatmarine) [inline OnPod/Node/Examples/Ethernet/main.cpp]: NODE_TYPE, all caps for consts

**Pull request merged**